test/parallel/test-http-socket-error-listeners.js

```
@@ -0,0 +1,84 @@
+'use strict';
+
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+const net = require('net');
+
+// This test sends an invalid character to a HTTP server and purposely
+// does not handle clientError (even if it sets an event handler).
```

As discussed in https://github.com/nodejs/node/pull/43587, in my opinion this is the issue, not a bug in Node.js core.

---

And I also agree.
But since there is an easy way to prevent people to be left in the dark when dealing with nasty utilizations like this, why shouldn't we help them?

---

Because we are just hiding a user error, making it harder to debug it.

---

But it is not different on how it is today. Since we add noop all the times, the error is hidden anyway. Am I wrong?

---

At least there is a warning now.

---

Fair enough. But the leak is completely out of context so it's hard for a user to see where is the problem.
Let's see what others think.

---

I agree. I'll add in the docs.

---

Done. What about now?

---

Can you move the commit to its own PR?

---

I would do the same for `EventEmitter.prototype.hasListener()` (move it to a separate PR without the HTTP changes) and nothing else.

---

Done. This PR now only affects EE. I'll send two other PRs, one for the docs and one for using hasListener in HTTP.